ci-operator/templates/cluster-launch-e2e-upgrade.yaml

```diff
@@ -0,0 +1,333 @@
+        oc create secret generic telemeter-client --from-literal=to=https://telemeter.svc.ci.openshift.org --from-literal=id=${NAMESPACE}-${JOB_NAME_HASH} --from-file=token=/tmp/cluster/telemeter-token --from-literal=salt=random-salt --from-file=rules=/tmp/rules --dry-run -o yaml | oc apply -n openshift-telemetry -f -
+        oc create -f https://raw.githubusercontent.com/openshift/telemeter/master/deploy/telemeter-client-openshift.yaml -n openshift-telemetry
+        set -e
+
+        # create the environment, but exit before we run tests
+        if [[ -n "${TEST_COMMAND-}" ]]; then
+          ${TEST_COMMAND}
+          exit 0
+        fi
+
+        # TODO: the test binary should really be a more structured command - most of these flags should be
+        #       autodetected from the running cluster.
+        # TODO: bump nodes up to 40 again
+        set -x
+        if [[ -n "${TEST_SUITE}" || -n "${TEST_FOCUS}" ]]; then
+          ginkgo -v -noColor -nodes=30 $( which extended.test ) -- \
+            -suite "${TEST_SUITE}" -ginkgo.focus="${TEST_FOCUS}" -ginkgo.skip="${TEST_SKIP}" \
+            -e2e-output-dir /tmp/artifacts -report-dir /tmp/artifacts/junit \
+            -test.timeout=2h ${PROVIDER_ARGS-} || rc=$?
+        fi
+        if [[ -n "${TEST_SUITE_SERIAL}" || -n "${TEST_FOCUS_SERIAL}" ]]; then
+          ginkgo -v -noColor -nodes=1 $( which extended.test ) -- \
+            -suite "${TEST_SUITE_SERIAL}" -ginkgo.focus="${TEST_FOCUS_SERIAL}" -ginkgo.skip="${TEST_SKIP_SERIAL}" \
+            -e2e-output-dir /tmp/artifacts -report-dir /tmp/artifacts/junit/serial \
+            -test.timeout=2h ${PROVIDER_ARGS-} || rc=$?
+        fi
+        exit ${rc:-0}
+
+    # Runs an install of a previous release
+    - name: setup
+      image: ${IMAGE_PREVIOUS_ANSIBLE}
+      volumeMounts:
+      - name: shared-tmp
+        mountPath: /tmp
+      - name: cluster-profile
+        mountPath: /usr/share/ansible/openshift-ansible/inventory/dynamic/injected
+      env:
+      - name: INSTANCE_PREFIX
+        value: ${NAMESPACE}-${JOB_NAME_HASH}
+      - name: TYPE
+        value: ${CLUSTER_TYPE}
+      - name: UPGRADE_VERSION
+        value: ${TEST_UPGRADE}
+      - name: IMAGE_PREVIOUS_FORMAT
+        value: ${IMAGE_PREVIOUS_FORMAT}
+      command:
+      - /usr/local/bin/entrypoint-provider
+      args:
+      - /bin/bash
+      - -c
+      - |
+        #!/bin/bash
+        set -euo pipefail
+
+        trap 'rc=$?; if [[ $rc -ne 0 ]]; then touch /tmp/shared/install; fi; exit $rc' EXIT
+        trap 'kill $(jobs -p); exit 0' TERM
+
+        ansible-playbook -vv \
+          -e openshift_test_repo=${TEST_PREVIOUS_RELEASE_RPM_REPO} -e 'oreg_url=${IMAGE_PREVIOUS_FORMAT}' \
+          playbooks/${CLUSTER_TYPE}/openshift-cluster/launch.yml
+
+        # If the cluster would be upgraded /tmp/shared/admin.kubeconfig needs to be moved
+        if [[ -n "${UPGRADE_VERSION:-}" ]]; then
+          mv /tmp/shared/admin.kubeconfig{,_}
```

Comment:
As implemented, regardless of whether an upgrade does or does not run, the test container may see this file start running once the initial setup is done

Reply:
Done, this must be the remainder from minor upgrade template

Reply:
Right, this would require a fix in openshift-ansible - the part which copies admin.kubeconfig to `/tmp` should be moved to the template instead.

Updated this template to use that, but I'm not entirely sure how would it work with dynamic inventories though